File: backend/secondbrain/src/main/java/uknowklp/secondbrain/api/note/dto/NoteSearchResponse.java
package uknowklp.secondbrain.api.note.dto;

import java.util.List;

import lombok.Builder;

// 노트 검색 응답 DTO (페이징 정보 포함)
@Builder
public record NoteSearchResponse(
	List<NoteSearchResult> results,
	long totalCount,
	int currentPage,
	int totalPages,
	int pageSize
) {
}

File: mobile_watch/mobile/src/main/java/com/example/secondbrain/data/model/NoteSearchResponse.kt
package com.example.secondbrain.data.model

import com.google.gson.annotations.SerializedName

// 노트 검색 응답 데이터 모델
data class NoteSearchResponse(
    @SerializedName("results")
    val results: List<NoteSearchResult>,

    @SerializedName("totalCount")
    val totalCount: Long,

    @SerializedName("currentPage")
    val currentPage: Int,

    @SerializedName("totalPages")
    val totalPages: Int,

    @SerializedName("pageSize")
    val pageSize: Int
)
